### PR TITLE
Log time series details for out-of-order samples in remote write receiver

### PR DESCRIPTION
**Pull request by rabenhorst**
Log time series details (tags and timestamp) for `ErrOutOfOrderSample`, `ErrOutOfBounds` and `ErrDuplicateSampleForTimestamp` errors in remote-write-receiver (write handler). This is useful for debugging out-of-order samples (e.g. https://github.com/prometheus/prometheus/issues/9218#issuecomment-926495597).

### CODE REVIEW

**Review by codesome — CHANGES_REQUESTED** [review]
This seems to be doing same thing twice. You can have `switch errors.Cause(err) {` in `ServeHTTP` instead of `switch err {` that is already there with the additional fields that you have added in the log.

**Comment by rabenhorst** [comment]
> This seems to be doing same thing twice. You can have `switch errors.Cause(err) {` in `ServeHTTP` instead of `switch err {` that is already there with the additional fields that you have added in the log.

Yes, but there I don't have the details of the out-of-order sample and can't log it. The idea is to log the sample that caused the out-of-order error for better debugging.

**Review by codesome — COMMENTED** [review]
Oh right, missed that part. Could you please inline `logAppendError()` where it's called? Since it's a single place, feels unnecessary to have it separated.

**Comment by rabenhorst** [comment]
> Oh right, missed that part. Could you please inline `logAppendError()` where it's called? Since it's a single place, feels unnecessary to have it separated.
Done.

File: storage/remote/write_handler.go [hunk 1]
```diff
@@ -98,6 +97,7 @@ func (h *writeHandler) write(ctx context.Context, req *prompb.WriteRequest) (err
 		for _, s := range ts.Samples {
 			_, err = app.Append(0, labels, s.Timestamp, s.Value)
 			if err != nil {
+				level.Error(h.logger).Log("msg", "Out of order sample from remote write", "err", err.Error(), "series", labels.String(), "timestamp", s.Timestamp)
```
Comment (codesome) [inline storage/remote/write_handler.go:100]:
```suggestion
			switch errors.Cause(err) {
			case storage.ErrOutOfOrderSample, storage.ErrOutOfBounds, storage.ErrDuplicateSampleForTimestamp:
				level.Error(h.logger).Log("msg", "Out of order sample from remote write", "err", err.Error(), "series", labels.String(), "timestamp", s.Timestamp)
			}
```

**Comment by rabenhorst** [comment]
Sry may bad only copied half the code. Should be fixed now.

**Review by codesome — APPROVED** [review]

**Comment by maxpain** [comment]
When it will be released?

**Pull request merged**